PSPs/drafts/psp-6.md

```diff
@@ -0,0 +1,2903 @@
+
+#### 1.7.1.3. Example
+
+Request:
+
+```bash
+curl -H "Content-Type: application/json" -d '{"id":1, "jsonrpc":"2.0", "method": "grandpa_roundState", "params":[]}' http://localhost:9933
+```
+
+Response (shortened):
+
+```json
+{
+    "jsonrpc": "2.0",
+    "result": {
+        "background": [],
+        "best": {
+            "precommits": {
+                "currentWeight": 0,
+                "missing": [
+                    "1342Tr5GgVmEFZB9EQh4CvBwZukc8b9wWkf1jWURRNx5DHd",
+                    "1878HXjkN7yHw63iKqhW5d7Q7MfhceaS5sZmADBW32WRHLq",
+                    "16gGqUnuCFPHHc6W34EznXe2FNcy6V8KzM9VymRF1EiKMKoG",
+                    "16ist66RDhSo3DoTQ3scCSWPvMCrVeDMa7uqmMLvsGHGcVtz",
+                    "16jY7XReiowJ2MwiF9nFKomCQPJfykwnBu3hqJM25FqYiyfC"
+                ]
+            },
+            "prevotes": {
+                "currentWeight": 0,
+                "missing": [
+                    "1342Tr5GgVmEFZB9EQh4CvBwZukc8b9wWkf1jWURRNx5DHd",
+                    "1878HXjkN7yHw63iKqhW5d7Q7MfhceaS5sZmADBW32WRHLq",
+                    "16gGqUnuCFPHHc6W34EznXe2FNcy6V8KzM9VymRF1EiKMKoG",
+                    "16ist66RDhSo3DoTQ3scCSWPvMCrVeDMa7uqmMLvsGHGcVtz",
+                    "16jY7XReiowJ2MwiF9nFKomCQPJfykwnBu3hqJM25FqYiyfC"
+                ]
+            },
+            "round": 8329,
+            "thresholdWeight": 157,
+            "totalWeight": 234
+        },
+        "setId": 238
+    },
+    "id": 1
+}
+```
+
+### 1.7.2. grandpa_proveFinality
+
+Prove finality for the provided block range. Returns `NULL` if there are no known finalized blocks in the range. If no authorities set is provided, the current one will be attempted.
+
+#### 1.7.2.1. Parameter
+
+* `HASH` - The block hash indicating the beginning of the range.
+* `HASH` - The block hash indicating the ending of the range.
+* `U64` - (OPTIONAL) The Authority Set ID.
+
+#### 1.7.2.2. Response
+
+* `HEX` - (OPTIONAL) The SCALE encoded array of finality proofs.
+
+#### 1.7.2.3. Example
+
+Request:
+
+```bash
+curl -H "Content-Type: application/json" -d '{"id":1, "jsonrpc":"2.0", "method": "grandpa_proveFinality", "params":["0xd8c6bfd22432e1fdf2fbe4e6f32fb650b7bb5a9c4c992d53129bd64882f50290", "0xa324b88c18fb7109af79f03de4eee88bd6207376a085898c82d67b7d1a35ac41", 244]}' http://localhost:9933
+```
+
+Response (shortened):
+
+```json
+{
+   "jsonrpc":"2.0",
+   "result":"0x044e8e00c93905add716002173aa62feb3...83666edbb4da7ae3573c962f710f6e8500",
+   "id":1
+}
+```
+
+### 1.7.3. grandpa_subscribeJustifications (pubsub)
+
+Returns the block most recently finalized by Grandpa, alongside side its justification.
+
+This endpoints communicates over the Websocket protocol (`grandpa_justifications` subscription).
+
+#### 1.7.3.1. Parameter
+
+None.
+
+#### 1.7.3.2. Response
+
+* `HEX` - The block itself.
+
+#### 1.7.3.3. Example
+
+Request:
+
+```json
+{
+    "id": 1,
+    "jsonrpc": "2.0",
+    "method": "grandpa_subscribeJustifications",
+    "params": []
+}
+```
```

Comment:
I'm not sure about this. Curl examples are quite useful since people can just copy paste those in their own environment (which runs Substrate/Polkadot, for example). With websockets it's not that easy. Personally I used [wscat](https://github.com/websockets/wscat).

Looks like there is a `-x` parameter which can be used as a single command. I can adjust the examples if you think this should be added, because you're right about readability.